server/app/services/program/predicate/LeafOperationExpressionNode.java

```diff
@@ -67,12 +67,12 @@ public PredicateExpressionNodeType getType() {
   public String toDisplayString(ImmutableList<QuestionDefinition> questions) {
     Optional<QuestionDefinition> question =
         questions.stream().filter(q -> q.getId() == questionId()).findFirst();
+    String displayValue =
+        question
+            .map(q -> comparedValue().toDisplayString(q))
+            .orElseGet(() -> comparedValue().value());
```

Comment:
This moves the fall back to value when question is empty to here instead of awkwardly in toDisplayString()